internal/gcsx/append_object_creator.go

```diff
@@ -125,7 +126,9 @@ func (oc *appendObjectCreator) Create(
 		MetadataMap[key] = value
 	}
 
-	MetadataMap[MtimeMetadataKey] = mtime.Format(time.RFC3339Nano)
+	if mtime != nil {
```

Comment:
Please correct me if I am wrong here but I believe this flow is not modified by local file changes. Why did we add this check?

Reply:
Earlier mtime is a value now its a pointer which can be nil. hence validation is required.